src/java/org/apache/cassandra/db/compression/CompressionDictionaryManager.java

```diff
@@ -283,7 +283,14 @@ public CompositeData getCompressionDictionary()
         if (compressionDictionary == null)
             return null;
 
-        return CompressionDictionaryDetailsTabularData.fromCompressionDictionary(keyspaceName, tableName, compressionDictionary);
+        try
+        {
+            return CompressionDictionaryDetailsTabularData.fromCompressionDictionary(keyspaceName, tableName, compressionDictionary);
+        }
+        finally
+        {
+            compressionDictionary.close();
+        }
```

Comment:
without doing it like this we are not closing anywhere so we would keep dictionary we just retrieved from disk dangling, not releasing it anywhere, then once that gets GCed it would leak.